src/viadot/sources/tm1.py

```diff
@@ -135,79 +125,89 @@ def get_dimensions_names(self) -> list:
         conn = self.get_connection()
         return conn.dimensions.get_all_names()
 
-    def get_hierarchies_names(self) -> list:
+    def get_hierarchies_names(self, dimension: str) -> list:
         """Get list of available hierarchies in TM1 dimension instance.
 
+        Args:
+            dimension (str): Dimension name.
+
         Returns:
             list: List containing available hierarchies names.
 
         """
-        if self.dimension is None:
-            msg = "Missing dimension name."
-            raise ValidationError(msg)
-
         conn = self.get_connection()
-        return conn.hierarchies.get_all_names(self.dimension)
+        return conn.hierarchies.get_all_names(dimension)
 
-    def get_available_elements(self) -> list:
+    def get_available_elements(self, dimension: str, hierarchy: str) -> list:
         """Get list of available elements in TM1 instance.
 
         Elements are extracted based on hierarchy and dimension.
 
+        Args:
+            dimension (str): Dimension name.
+            hierarchy (str): Hierarchy name.
+
         Returns:
             list: List containing available elements names.
 
         """
-        if (self.dimension or self.hierarchy) is None:
+        if (dimension or hierarchy) is None:
```

Comment:
`or` will always resolve to a boolean, so this expression would never be `True` (neither `True` or `False` is None)
```suggestion
        if dimension is None or hierarchy is None:
```